docs/source/contribution_guidelines.rst

```diff
@@ -0,0 +1,127 @@
+Contribution Guidelines
+***********************
+
+Being open, we expect contributions from various walks of life. The below guidelines help in keeping the code base uniform irrespective of the person contributing to it.
+
+Coding
+------
+* Follow the PEP 8 proposal. Specifically, for the below mentioned areas
+    - `Naming Conventions`_
+    - `Programming recommendations`_
+    - `Whitespaces`_
+    - Using code comments sparsely but if required then `PEP8`_
+
+* Docstrings is mandatory, it needs to be compact and informative. It is required for each level i.e. functions, methods, classes and modules.
+    - Follow the guidelines from `PEP257`_
+    - It should be considered as a living document i.e. change them in accordance with the code modifications.
+    - All parameters must be documented
+    - Return values must be documented
+    - Document the exceptions raised if any
+
+* Import modules and its attributes
+    - Group them with a separation in the below order,
+        - Standard library
+        - Third-party library
+        - Local library
+        - Local module/package
+    - Use absolute imports
+    - Avoid relative import
+    - Import based on requirement and use from statement wherever applicable
+    - `PEP328`_ is a good convention that can be used
+
+* Avoid using ``*args **kwargs`` variable length of parameters,
+
+    .. code-block:: python
+
+        def main(*args, **kwargs):
+        ....
+
+* Prefer to use key/value pairs in methods over arguments
+    - Good: 	``x = wait_till(timeout=300,polling=10)``
+    - Bad:      ``x = wait_till(300, 10)``
+
+* Write unit tests for new code that is applicable to the framework.
+
+
+Testing
+-------
+- Unit test the changes
+- Run system tests if required
+- Ensure to run tox before commit
+
+Best Practices
+--------------
+Let’s apply some of the industry standard practices during our development
```

Comment:
Over here, the following best practices for development have been chosen from the vast set of them.